Fix for issue 9053: highlights wrong characters

Anqiii-hub opened the pull request:
<!-- 
Describe the changes you have made here: what, why, ... 
Link issues that are fixed, e.g. "Fixes #333".
If you fixed a koppor issue, link it, e.g. "Fixes https://github.com/koppor/jabref/issues/47".
The title of the PR must not reference an issue, because GitHub does not support autolinking there.
-->
Fixes #9053 

<!-- 
- Go through the list below. Please don't remove any items.
- [x] done; [ ] not done / not applicable
-->

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

HoussemNasri commented:
Please include a screenshot of the merge dialog with your changes in the PR description.

Anqiii-hub commented:
> Please include a screenshot of the merge dialog with your changes in the PR description.

Just added, thanks for reminding. For more detailed description, I updated in the issue.

ThiloteE commented:
**This PR causes a regression:**

I set the keyword separator to `;` via `options > preferences > groups` and did a test.

It turns out that contrary to expectations, adding an entry to a group will **NOT** use `;` as keyword separator and instead will use `,` (All the `;` in the following screenshot were added manually). I tested in original JabRef and there, using `;` was possible. Hence, this PR breaks the keyword separator preference.

 Furthermore, JabRef will not detect keyword separation at all during "show diff" as you can see in this screenshot:

![grafik](https://user-images.githubusercontent.com/73715071/199298557-750d0875-c325-47d3-982d-ffb5c9075af5.png)

This is a regression, because in the following screenshot you will see behaviour for `;` as keyword separator in 
JabRef 5.8--2022-10-31--e7c8535
Linux 5.4.0-131-generic amd64 
Java 18.0.2.1 
JavaFX 19+11

![grafik](https://user-images.githubusercontent.com/73715071/199299618-fab79fd3-a149-4429-bddd-1646045bead1.png)

**Positive test results:**

Using `,` as keyword separator worked flawlessly :-)

Siedlerchr commented:
I think to move forward here: Instead of hard coding a comma, the character from  GroupPreferences: getKeywordSeparator should be used
@Anqiii-hub  I would be great if you could change this. You would need to pass the separator as a constructor argument or simlar

Siedlerchr commented:
@ThiloteE  I managed to read the separator out of the preferences now

<img width="1704" alt="grafik" src="https://user-images.githubusercontent.com/320228/201742168-0e5df871-6112-4d54-912e-8f217ee1bb96.png">

Sorry that PR escalated a bit ;)

ThiloteE commented:
Works with `,`

![image](https://user-images.githubusercontent.com/73715071/203151096-d1264ee3-1375-4f54-a34f-c89e3fdb2af9.png)
![image](https://user-images.githubusercontent.com/73715071/203150211-f18617ee-5874-41bf-a6fc-08ffe67580ac.png)

Also works with `;` when set in preferences

![image](https://user-images.githubusercontent.com/73715071/203150536-a7c6ab8d-ac01-41ab-a859-e4860fd47123.png)
![image](https://user-images.githubusercontent.com/73715071/203150505-468daad3-0d58-40e4-81f5-dc8f3b2c0855.png)

calixtus commented on src/main/java/org/jabref/gui/mergeentries/newmergedialog/ThreeWayMergeView.java:
```diff
@@ -73,11 +74,19 @@ private void initializeToolbar() {
     private void updateDiff() {
         if (toolbar.isShowDiffEnabled()) {
-            fieldRows.forEach(row -> row.showDiff(new ShowDiffConfig(toolbar.getDiffView(), toolbar.getDiffHighlightingMethod())));
+            for (int i = 0; i < fieldRows.size(); i++) {
+                FieldRowView row = fieldRows.get(i);
```
Seems to be the only place where i is needed. So why not forEach?

calixtus reviewed (dismissed):
Just one question, everything else looks good to me
